lib/mobius/actions/events.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Mobius.Actions.Events do
+  @moduledoc """
+  Functions related to events
+  """
+
+  alias Mobius.Services.EventPipeline
+
+  @doc """
+  Subscribes the calling process to a set of (or all) events
+
+  Subscribing multiple times with overlapping events will result
+  in duplicate events. As such, it is discouraged to subscribe
+  multiple times from the same process since this may cause
+  performance degradation
+
+  If an empty list (the default) is given, subscribes to all events
+
+  The caller process is then sent events whose name is in the set
+  as messages in the format `{event_name, event_data}`
+  where `event_name` is the name of the event as an UPPER_CASE atom
+  such as `:CHANNEL_CREATE` and where `event_data` is a struct
+  containing all the information given by the event
```

Comment:
```suggestion
  The caller process is then sent events whose name is in the set
  as messages in the format `{event_name, event_data}`
  where `event_name` is the name of the event as an UPPER_CASE atom
  such as `:CHANNEL_CREATE` and `event_data` is a struct
  containing all the information given by the event
```

Also, do we actually need to send the event name outside of the struct? Isn't the struct going to contain this information anyway?

Reply:
After reading the rest of the code, I see that the events don't actually contain a struct, but rather a map. Are there plans to create an actual struct?

Reply:
Yes, there's plans to translate the maps into structs. We could make a wrapper struct to contain both the event name and the data, but I don't really see an advantage to doing that when they still need to be split up for things like `subscribe/1` and parsing.

The event data's struct isn't unique per event name, for example all three of [`Channel Create`, `Channel Update`, and `Channel Delete` have a `Channel` struct as their data](https://discord.com/developers/docs/topics/gateway#channel-create). So the parsers are mapping [resources](https://discord.com/developers/docs/resources/channel#channel-object-channel-structure) and the event name is used to know which resource we received.

Reply:
Also the parsing is gonna be reused for the HTTP responses, so adding a struct here would really only be a wrapper around `name` and `data`. Adding `name` inside the `data` wouldn't work for the HTTP responses and wouldn't work for nested structures (like users inside channels inside guilds)